compiler/util/ir_data_utils.py

```diff
@@ -12,51 +12,449 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 
+"""Provides a helpers for working with IR data elements.
+
+builder
+-------
+Instead of:
+```
+def set_function_name_end(function: Function):
+  if not function.function_name:
+    function.function_name = Word()
+  if not function.function_name.source_location:
+    function.function_name.source_location = Location()
+  word.source_location.end = Position(line=1,column=2)
+```
+
+We can do:
+```
+def set_function_name_end(function: Function):
+  builder(function).function_name.source_location.end = Position(line=1,
+  column=2)
+```
+
```

Comment:
Maybe this could be a follow up? I'm probably not the best person to articulate the historical motivations for this behavior nor whether it's intended to act like other languages (nor do I want to open the door to "bugs" for not faithfully implementing their functionality).

Reply:
I'll add some sort of nod to autovivication et al. Annotating with `builder` at least calls out where we're doing this, I think a follow up might just remove builder but that was a wider ranging change than I want to take on here.